### DISCUSSION TITLE
OAuth intergration

### DISCUSSION THREAD

**Discussion by Ortovoxx** [payloadcms/payload#634] [Feature Requests & Ideas]
I looked through the docs and the repo and there doesn't seem to be any mention of OAuth intergration so I can use SSO with payload instead of individually assigning admins usernames and passwords. Am I missing something and this already exists or is it a planned feature?

I feel this is something that a plugin could add so would happily write a plugin to add this functionality if others are interested?

**Comment by jmikrut** [comment]
Hey @Ortovoxx — 

We do currently offer SSO support via our Enterprise packages, but there is no public implementation thus far. It could potentially be possible via plugins, but it's possible that outside of our enterprise support, we may have to add a few features here and there to properly build a first-class solution.

If you're up for taking a shot at a plugin-based implementation, or even a PR with features you need to make an SSO plugin possible, that would be awesome! Otherwise, if you're interested in hearing about our Enterprise SSO support, reach out to us at [info@payloadcms.com](mailto:info@payloadcms.com) and we can talk further.

**Reply by Ortovoxx** [reply]
Would definitely be interested in adding this functionality. Not super familiar with payload but I'll have a play over the next few months and see what I can get done in terms of a plugin or a PR!

Unfortunately, I'm not looking for Enterprise support as it's only for a small university website with a few users but the ability to have Azure AD SSO ( with our university accounts ) would be so much easier than having to give everyone a username and password as everyone would already be on the system anyway through mongo.

**Reply by robclancy** [reply]
I knew payloadcms was too good to be true.

**Comment by adam-mrozik** [comment]
@Ortovoxx  Hey! Just reviving the topic if anyone is working on this kind of plugin or is it planned for payload CMS public iplementation

**Reply by Ortovoxx** [reply]
I'm not currently working on it no. Not sure what the official line is for this however. Seems people are interested in this though, not sure the scale of work that would be required. Maybe a PR or perhaps a plugin?

**Comment by adam-mrozik** [comment]
https://github.com/Genially/passport-google-one-tap

This one seems similar enough. I am trying to adapt it to my internal needs, but seems like a start

**Comment by thgh** [comment]
Hi, I just published https://github.com/thgh/payload-plugin-oauth, what do you think?

**Reply by Ortovoxx** [reply]
Looks fantastic! Haven't had the chance to implement it but had skim through and looks to fit the bill. Nice one!

**Comment by hunghvu** [comment]
>As of Payload 1.0.0, you can add additional authentication strategies to Payload easily by passing them to your collection's auth.strategies array.
>
>Behind the scenes, Payload uses PassportJS to power its local authentication strategy, so most strategies listed on the PassportJS website will work seamlessly. Combined with adding custom components to the admin panel's Login view, you can create advanced authentication strategies directly within Payload.

Source: https://payloadcms.com/docs/authentication/config

[Passport](https://www.passportjs.org/) supports a variety of OAuth strategies even for Enterprise use cases. I have not tried it out yet, is this not applicable to your use cases?

**Reply by thgh** [reply]
TLDR: The oAuth strategy immediately redirects the user when not logged in, leaving the user no choice which provider to sign in with.
Full version: https://discord.com/channels/967097582721572934/1017917726007763024/1017938938436010044
